ruby_enumerable_methods.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    select_arr = []
+    for item in self do
+      select_arr << item if yield(item)
+    end
+    select_arr
+  end
+
+  def my_all?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break unless pass
+    end
+    pass
+  end
+
+  def my_any?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break if pass
+    end
+    pass
+  end
+
+  def my_none?
+    pass = false
+    for item in self do
+      pass = !yield(item)
+      break unless pass
+    end
+    pass
+  end
+
+  def my_count
+    length
+  end
+
+  def my_map(my_proc = nil)
+    new_arr = []
+    return self unless my_proc || block_given?
+
+    for item in self do
+      block_given? ? new_arr << yield(item) : new_arr << code.call(item)
+    end
+    new_arr
+  end
+
+  def my_inject(num = nil)
+    accumulator = num.nil? ? first : num
+    my_each { |i| accumulator = yield(accumulator, i) }
+    accumulator
+  end
+end
```

Comment:
- enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: expect(array.my_inject(&operation)).to eq array.inject(&operation)

- enumerables my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
      Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)